dimod/generators/satisfiability.py

```diff
@@ -59,25 +67,39 @@ def random_kmcsat(variables: typing.Union[int, typing.Sequence[dimod.typing.Vari
     that consists in satisfying a number of conjunctive
     clauses of ``k`` literals (variables, or their negations).
     Each clause should encode a max-cut problem over the clause literals.
+    
+    Each clause contributes -:code:`k//2' to the energy when the clause is 
```

Comment:
you're right, I forgot to adjust the offset/scale. You can feel free to add the code to do that so that satisfying solutions have 0 energy and violations (in the nae3sat case) add energy of 1. Or I can do that as a followup

Reply:
Current definition matches the literature we are citing, and has some other advantages (Ising models without a constant is fairly standard the main one). So I'll leave as is.